zebra/zebra_vxlan.c

```diff
@@ -7682,6 +7682,54 @@ int zebra_vxlan_handle_kernel_neigh_update(struct interface *ifp,
 	return zvni_remote_neigh_update(zvni, ifp, ip, macaddr, state);
 }
 
+static int32_t
+zebra_vxlan_remote_macip_helper(bool add, struct stream *s, vni_t *vni,
+				struct ethaddr *macaddr, uint16_t *ipa_len,
+				struct ipaddr *ip, struct in_addr *vtep_ip,
+				uint8_t *flags, uint32_t *seq)
+{
+	uint16_t l = 0;
+
+	/*
+	 * Obtain each remote MACIP and process.
+	 * Message contains VNI, followed by MAC followed by IP (if any)
+	 * followed by remote VTEP IP.
+	 */
+	memset(ip, 0, sizeof(*ip));
+	STREAM_GETL(s, *vni);
+	STREAM_GET(macaddr->octet, s, ETH_ALEN);
+	STREAM_GETL(s, *ipa_len);
+
+	if (*ipa_len) {
+		if (*ipa_len != IPV4_MAX_BYTELEN
+		    && *ipa_len != IPV6_MAX_BYTELEN) {
+			if (IS_ZEBRA_DEBUG_VXLAN)
+				zlog_debug(
+					"ipa_len *must* be %d or %d bytes in length not %d",
+					IPV4_MAX_BYTELEN, IPV6_MAX_BYTELEN,
+					*ipa_len);
+			goto stream_failure;
+		}
+		ip->ipa_type =
+			(*ipa_len == IPV4_MAX_BYTELEN) ? IPADDR_V4 : IPADDR_V6;
```

Comment:
This works, but I think this might be simpler & clearer:
```c
if (*ipa_len == IPV4_MAX_BYTELEN)
   ip->ipa_type = IPADDR_V4;
else if (*ipa_len == IPV6_MAX_BYTELEN)
   ip->ipa_type = IPADDR_v6;
else {
   debug();
   goto stream_failure;
}
```

Reply:
fixed